lib/packExternalModules.js

```diff
@@ -226,19 +268,35 @@ module.exports = {
       const packageJson = this.serverless.utils.readFileSync(packageJsonPath);
       const packageSections = _.pick(packageJson, sectionNames);
       if (!_.isEmpty(packageSections)) {
-        this.options.verbose &&
-          this.serverless.cli.log(`Using package.json sections ${_.join(_.keys(packageSections), ', ')}`);
+        if (this.log) {
+          this.log.verbose(`Using package.json sections ${_.join(_.keys(packageSections), ', ')}`);
+        } else {
+          this.options.verbose &&
+            this.serverless.cli.log(`Using package.json sections ${_.join(_.keys(packageSections), ', ')}`);
+        }
       }
 
       // Get first level dependency graph
-      this.options.verbose && this.serverless.cli.log(`Fetch dependency graph from ${packageJsonPath}`);
+      if (this.log) {
+        this.log.verbose(`Fetch dependency graph from ${packageJsonPath}`);
+      } else {
+        this.options.verbose && this.serverless.cli.log(`Fetch dependency graph from ${packageJsonPath}`);
+      }
```

Comment:
Can't we have an _util_ method which handle the log? Or do you prefer to keep all those `if` to ease the removal later (when we'll drop Serverless v2)?

Something like:

```js
function swLog ({ options, v3log, log, level, message }) {
  if (v3log) {
    this.log[level](message);
  } else {
    if (level === 'verbose') {
      options.verbose && log(`INFO: ${message}`);
    } else {
      log(`${level.toUpperCase()}: ${message}`);
    }
  }
}
```

And then:
```suggestion
      swLog ({
        options: this.options,
        v3log: this.log,
        log: this.serverless.cli.log,
        level: 'verbose',
        message: `Fetch dependency graph from ${packageJsonPath}`
      });
```

Reply:
Thinking about it, it might add useless complexity and keeping `if`/`else` might be better when we'll drop v2

Reply:
Yes, I think this `swLog` looks even more verbose and it's unclear for readers.

Ideally we should drop old logging format once Framework v3 is released, and mark plugin that it's compatible only with v3